lsp-mode.el

```diff
@@ -5203,18 +5238,52 @@ It will show up only if current point has signature help."
         :context `( :diagnostics ,(lsp-cur-line-diagnostics)
                     ,@(when kind (list :only (vector kind))))))
 
-(defun lsp-code-actions-at-point (&optional kind)
-  "Retrieve the code actions for the active region or the current line."
-  (lsp-request "textDocument/codeAction" (lsp--text-document-code-action-params kind)))
+(defun lsp-code-actions-at-point (&optional kind enabled)
+  "Retrieve the code actions for the active region or the current line.
+If ENABLED is specified, only return code actions that are not
+disabled."
+  (--doto (lsp-request "textDocument/codeAction" (lsp--text-document-code-action-params kind))
+    (when enabled
+      (cl-callf2 seq-remove #'lsp:code-action-disabled? it))))
 
 (defun lsp-execute-code-action-by-kind (command-kind)
   "Execute code action by name."
-  (if-let ((action (->> (lsp-get-or-calculate-code-actions command-kind)
-                        (-filter (-lambda ((&CodeAction :kind?))
-                                   (and kind? (equal command-kind kind?))))
-                        lsp--select-action)))
-      (lsp-execute-code-action action)
-    (signal 'lsp-no-code-actions '(command-kind))))
+  (let* ((all-actions (->> (lsp-get-or-calculate-code-actions command-kind)
+                           (-filter (-lambda ((&CodeAction :kind?))
+                                      (and kind? (equal command-kind kind?))))))
+         (enabled-actions (-remove #'lsp:code-action-disabled? all-actions))
+         (action (lsp--select-action (or enabled-actions all-actions))))
+    (lsp-execute-code-action action)))
+
+(defun lsp-execute-code-action-by-type (kind &optional enabled)
+  "Execute a code action with a given base KIND.
+When ENABLED is given, filter out all disabled code actions. This
+is in contrast with the spec's recommended behavior."
+  (->> (lsp-get-or-calculate-code-actions kind enabled)
+       (-filter (-lambda ((&CodeAction :kind?))
+                  (and kind? (string-match-p (format "\\`%s\\(\\.\\|\\'\\)"
+                                                     (regexp-quote kind))
+                                             kind?))))
+       lsp--select-action
+       lsp-execute-code-action))
+
+(defun lsp-quickfix ()
```

Comment:
There's `lsp-make-interactive-code-action` macro which use `lsp-execute-code-action-by-kind`, do you want to provide similar one for `lsp-execute-code-action-by-type`?